pallets/storage-provider/src/deadline.rs

```diff
@@ -745,6 +754,29 @@ where
             self.fault_declaration_cutoff,
         )
     }
+
+    /// Returns the next deadline that has not yet elapsed.
+    ///
+    /// If the current deadline has not elapsed yet then it returns the current deadline.
+    pub fn next_not_elapsed(self) -> Result<Self, GeneralPalletError> {
+        if !self.has_elapsed() {
+            return Ok(self);
+        }
+
+        // has elapsed, advance by some multiples of w_post_proving_period
```

Comment:
Some? I read `next` and think it's just one.

Maybe there's a better way of writing this

Reply:
This comment is unnecessary, I removed it.

It's not just one, cause if for some reason there is:
`[self.open_at, self.close_at] = [100, 120]` and current block = 162, then the next proving period is not `160, 220`, but `240, 260`.